rest/diagnostic_doc_api.go

```diff
@@ -12,9 +12,23 @@ package rest
 
 import (
 	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/channels"
 	"github.com/couchbase/sync_gateway/db"
 )
 
+type SyncFnDryRun struct {
+	Channels  base.Set           `json:"Channels"`
+	Access    channels.AccessMap `json:"Access"`
+	Roles     channels.AccessMap `json:"Roles"`
```

Comment:
Sync Function execution also returns 'expiry' - I think that would be useful information for the dry run.